.github/workflows/ci.yml

```
@@ -0,0 +1,164 @@
+          echo "RabbitMQ_uri=host=$ip" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf-8 -Append
+         
+          $uri = "http://" + $ip + ":15672/api/health/checks/virtual-hosts"
+          $tries = 0
+          do
+          {
+            $response = curl $uri -u guest:guest | ConvertFrom-Json
+            $tries++
+            if(!$response.status)
+            {
+              Write-Output "No response, retrying..."
+              Start-Sleep -m 5000
+            }
+          } until (($response.status) -or ($tries -ge 10))
+          if($response.status -ne "ok")
+          {
+            Write-Output "Failed to connect after 10 attempts";
+            exit 1
+          }
+      - name: Setup RavenDB4
+        if: contains(matrix.engines, 'RavenDB4')
+        id: setup-ravendb4
+        shell: pwsh
+        run: |
+          $hostInfo = curl -H Metadata:true "169.254.169.254/metadata/instance?api-version=2017-08-01" | ConvertFrom-Json
+          $region = $hostInfo.compute.location
+          
+          $hostname = "psw-ravendb-$(Get-Random)"
+          echo "::set-output name=hostname::$hostname"
+          echo "Creating RavenDB container $hostname in $region (This can take a while.)"
+          $details = az container create --image ravendb/ravendb:4.2-ubuntu-latest  --name $hostname --location $region --dns-name-label $hostname --resource-group GitHubActions-RG --cpu 4 --memory 16 --ports 8080 38888 --ip-address public -e RAVEN_ARGS='--ServerUrl=http://0.0.0.0:8080 --Setup.Mode=None --License.Eula.Accepted=true --Security.UnsecuredAccessAllowed=PublicNetwork' | ConvertFrom-Json
+          $fqdn=$details.ipAddress.fqdn
+          echo "::add-mask::$fqdn"
+          echo "Tagging container image"
+          $dateTag = "Created=$(Get-Date -Format "yyyy-MM-dd")"
+          $ignore = az tag create --resource-id $details.id --tags Package=TimeoutMigrationTool RunnerOS=${{ runner.os }} $dateTag
+          echo "CommaSeparatedRavenClusterUrls=http://$($fqdn):8080" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf-8 -Append
+      - name: Install SQL Server
+        if: contains(matrix.engines, 'SqlServer')
+        uses: Particular/install-sql-server-action@main
+        with:
+          connection-string-env-var: SQLServerConnectionString
+          catalog: nservicebus
+      - name: Run tests
+        if: ${{ !contains(matrix.engines, 'RabbitMQ') }}
+        uses: Particular/run-tests-action@group-output-by-test-project
+      - name: Run tests with RabbitMQ
+        if: contains(matrix.engines, 'RabbitMQ')
+        uses: Particular/run-tests-action@group-output-by-test-project
+        with:
+          reset-script: dotnet run --project src/targets
+      - name: Teardown infrastructure
+        if: ${{ always() }}
+        shell: pwsh
+        run: |
+          if ('${{ matrix.engines }}' -like 'RabbitMQ') {
+            echo "Removing container ${{ steps.setup-rabbitmq.outputs.name }}"
```

Output name is `hostname` above
```suggestion
            echo "Removing container ${{ steps.setup-rabbitmq.outputs.hostname }}"
```